t/24-worker-webui-connection.t

```diff
@@ -315,35 +321,37 @@ qr/502 response: some timeout \(remaining tries: 2\).*502 response: some timeout
         $fake_error->{message}                 = 'some error';
         $fake_error->{code}                    = 500;
         $web_ui_supposed_to_be_considered_busy = undef;
-        combined_like(
-            sub {
-                $client->send(@send_args, tries => 2);
-                Mojo::IOLoop->start;
-            },
+        send_once(\@send_args,
             qr/500 response: some error \(remaining tries: 1\).*500 response: some error \(remaining tries: 0\)/s,
-            'attempts logged',
-        );
+            undef, tries => 2);
         is($fake_ua->start_count, 2, 'tried 2 times');
         is($callback_invoked,     1, 'callback invoked exactly one time');
         is($retry_delay_invoked,  1, 'retry delay queried');
     };
 
-    subtest 'no retry after 404' => sub {
+    subtest 'no retry after 4XX (with exceptions)' => sub {
         $callback_invoked = $retry_delay_invoked = 0;
         $fake_ua->start_count(0);
-        $fake_error->{message} = 'Not found';
-        $fake_error->{code}    = 404;
-        combined_like(
-            sub {
-                $client->send(@send_args, tries => 3);
-                Mojo::IOLoop->start;
-            },
-            qr/404 response: Not found \(remaining tries: 0\)/,
-            '404 logged',
+        my %codes_4xx = (
+            400 => 'Not found',
+            401 => 'Unauthorized',
+            403 => 'Forbidden',
+            404 => 'Not found',
+            418 => 'I\'m a teapot',
+            426 => 'Upgrade Required',
```

Comment:
Instead of 418 and 426 I'd rather test 499. The bounds are likely more interesting than random values in between (even though these might not be official HTTP status codes).

Reply:
But I like 418 :)

Reply:
Me too, 418 must stay! :shipit:

Reply:
so I did not change this.